packages/shared/src/contexts/PushNotificationContext.tsx

```diff
@@ -54,10 +54,23 @@ export function PushNotificationContextProvider({
   const onSourceChange = useCallback((source) => {
     notificationSourceRef.current = source;
   }, []);
-  const subscriptionCallbackRef = useRef<SubscriptionCallback>();
   const [isSubscribed, setIsSubscribed] = useState(false);
   const { user } = useAuthContext();
   const { trackEvent } = useAnalyticsContext();
+  const subscriptionCallbackRef = useRef<SubscriptionCallback>(
+    (isSubscribedNew, source, existing_permission) => {
+      if (isSubscribedNew) {
+        trackEvent({
+          event_name: AnalyticsEvent.ClickEnableNotification,
+          extra: JSON.stringify({
+            origin: source || notificationSourceRef.current,
+            permission: 'granted',
+            ...(existing_permission && { existing_permission }),
+          }),
+        });
+      }
+    },
+  );
```

Comment:
You are going, to need to add line of `subscriptionCallbackRef.current = () => ...`, because `useRef` initial value is set only first time. You can see it in `useMutationSubscription`.

Reply:
Ahhh. use ref. Yeah, good catch.